### PR TITLE
odd-even implemented

### PR DESCRIPTION
**Pull request by Vilax**
This PR depends on ...

https://github.com/scipion-em/scipion-em-motioncorr/pull/81

 and

https://github.com/scipion-em/scipion-em-tomo/pull/337

### CODE REVIEW

File: imod/protocols/protocol_tomoNormalization.py [hunk 1]
```diff
@@ -234,16 +245,40 @@ def generateOutputStackStep(self, tsObjId):
...
 
+        oddEvenOutput = [[], []]
+
         if runNewstack:
             Plugin.runImod(self, 'newstack', argsNewstack % paramsNewstack)
 
+            if self.processOddEven and tomo.hasHalfMaps():
+                oddFn, evenFn = tomo.getHalfMaps().split(',')
+                paramsNewstack['input'] = oddFn
+                oddEvenOutput[0] = os.path.join(extraPrefix, tomo.getTsId() + EXT_MRC_ODD_NAME)
```
Comment (pconesa) [inline imod/protocols/protocol_tomoNormalization.py:256]: centralize?

File: imod/protocols/protocol_tsNormalization.py [hunk 2]
```diff
@@ -277,6 +300,12 @@ def generateOutputStackStep(self, tsObjId):
                 newTi.setTransform(None)
 
             newTi.setAcquisition(tiltImage.getAcquisition())
+            if self.processOddEven:
```
Comment (pconesa) [inline imod/protocols/protocol_tsNormalization.py:303]: and ts.hasOddEven()?

File: imod/protocols/protocol_doseFilter.py [hunk 3]
```diff
@@ -163,14 +186,23 @@ def createOutputStep(self, tsObjId):
...
+        ih = ImageHandler()
+
         for index, tiltImage in enumerate(ts):
             newTi = tomoObj.TiltImage()
             newTi.copyInfo(tiltImage, copyId=True, copyTM=True)
             newTi.setAcquisition(tiltImage.getAcquisition())
-            newTi.setLocation(index + 1, (os.path.join(extraPrefix,
-                                                       tiltImage.parseFileName())))
-
+            if self.processOddEven:
```
Comment (pconesa) [inline imod/protocols/protocol_doseFilter.py:195]: and ts.hasOddEven()

File: imod/protocols/protocol_ctfCorrection.py [hunk 4]
```diff
@@ -226,6 +252,12 @@ def createOutputStep(self, tsObjId):
             newTi.setLocation(index + 1,
                               (os.path.join(extraPrefix,
                                             tiltImage.parseFileName())))
+            if self.processOddEven:
```
Comment (pconesa) [inline imod/protocols/protocol_ctfCorrection.py:255]: and ts.hasOddEven()

File: imod/protocols/protocol_applyTransformationMatrix.py [hunk 5]
```diff
@@ -151,6 +175,11 @@ def generateOutputStackStep(self, tsObjId):
                 acq.setTiltAxisAngle(0.)
                 newTi.setAcquisition(acq)
                 newTi.setLocation(index, (os.path.join(extraPrefix, tiltImage.parseFileName())))
+                if self.processOddEven:
```
Comment (pconesa) [inline imod/protocols/protocol_applyTransformationMatrix.py:178]: and ts.hasOddEven()

File: imod/protocols/protocol_xRaysEraser.py [hunk 6]
```diff
@@ -172,13 +196,21 @@ def createOutputStep(self, tsObjId):
...
+        ih = ImageHandler()
+
         for index, tiltImage in enumerate(ts):
             newTi = tomoObj.TiltImage()
             newTi.copyInfo(tiltImage, copyId=True, copyTM=True)
             newTi.setAcquisition(tiltImage.getAcquisition())
             newTi.setLocation(index + 1,
                               (os.path.join(extraPrefix,
                                             tiltImage.parseFileName())))
+            if self.processOddEven:
```
Comment (pconesa) [inline imod/protocols/protocol_xRaysEraser.py:208]: and ts.hasOddEven()

**Review by pconesa — COMMENTED** [review]
Centralize code?
check param + hasOddEven.

**Review by azazellochg — APPROVED** [review]
I've run a simple workflow and all seem working! Thanks @Vilax !

**Pull request merged**